<?xml version="1.0"?>

<robot name="puzzle_drone">

   <link name="base_link">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="package://link_act/meshes/Puzzlebot_Jetson_Lidar_Edition_Base.stl"/>  
            </geometry>
            <material name="yellow">
                <color rgba="0.8 0.8 0.05 1.0"/>
            </material>
        </visual>
    </link>


   <link name="wheel_r_link">
        <visual>
            <origin xyz="0 0 0" rpy="1.57 0 0"/>
            <geometry>
                <mesh filename="package://link_act/meshes/Puzzlebot_Wheel.stl"/>
            </geometry>
            <material name="yellow">
                <color rgba="0.8 0.8 0.05 1.0"/>
            </material>
        </visual>
    </link>

    <link name="wheel_l_link">
        <visual>
            <origin xyz="0 0 0" rpy="1.57 0 0"/>
            <geometry>
                <mesh filename="package://link_act/meshes/Puzzlebot_Wheel.stl"/>
            </geometry>
            <material name="yellow">
                <color rgba="0.8 0.8 0.05 1.0"/>
            </material>
        </visual>
    </link>

    <link name="caster_link">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="package://link_act/meshes/Puzzlebot_Caster_Wheel.stl"/>
            </geometry>
            <material name="yellow">
                <color rgba="0.8 0.8 0.05 1.0"/>
            </material>
        </visual>
    </link>


    <joint name="wheel_r_joint" type="continuous">
        <parent link="base_link"/> 
        <child link="wheel_r_link"/> 
        <origin xyz="0.052 -0.095 -0.0025" rpy="0 0 0"/>
        <axis xyz="0 1 0"/> 
    </joint>

    <joint name="wheel_l_joint" type="continuous">
        <parent link="base_link"/> 
        <child link="wheel_l_link"/> 
        <origin xyz="0.052 0.095 -0.0025" rpy="0 0 0"/>
        <axis xyz="0 1 0"/> 
    </joint>

    <joint name="caster_joint" type="fixed">
        <parent link="base_link"/> 
        <child link="caster_link"/> 
        <origin xyz="-0.095 0.0 -0.03" rpy="0 0 0"/>
    </joint>

</robot>